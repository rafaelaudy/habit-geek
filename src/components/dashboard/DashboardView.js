import React, { useState, Fragment } from "react";
import { useTranslation } from "react-i18next";

import HabitTable from "../habits/HabitTable";
import DashboardWeekHeader from "./DashboardWeekHeader";

const DashboardView = ({
  username,
  currentHabits,
  previousHabits,
  currentWeek,
  previousWeek,
  onAddNewHabit,
  onUpdateHabit,
  toggleDayHabit
}) => {
  const [isCurrentWeek, setIsCurrentWeek] = useState(true);
  const { t } = useTranslation();

  let tUsername = username ? " " + username : username;

  return (
    <div className="large-size-container">
      <h2 className="mb-3">{t("my-habits-title", { username: tUsername })}</h2>

      <DashboardWeekHeader
        week={isCurrentWeek ? currentWeek : previousWeek}
        isCurrentWeek={isCurrentWeek}
        hasPreviousWeek={previousWeek}
        toggleWeek={setIsCurrentWeek}
      ></DashboardWeekHeader>

      <HabitTable
        habits={isCurrentWeek ? currentHabits : previousHabits}
        week={isCurrentWeek ? currentWeek : previousWeek}
        toggleDayHabit={toggleDayHabit}
        onUpdateHabit={onUpdateHabit}
      ></HabitTable>

      {isCurrentWeek ? (
        <Fragment>
          <hr />
          <div className="d-flex justify-content-center">
            <button className="btn btn-primary btn-lg" onClick={onAddNewHabit}>
              {t("my-habits-new-habit")}
            </button>
          </div>
        </Fragment>
      ) : null}
    </div>
  );
};

export default DashboardView;
