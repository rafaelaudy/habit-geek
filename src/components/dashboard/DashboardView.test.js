import React from "react";
import { shallow, mount } from "enzyme";
import DashboardView from "./DashboardView";

const props = {
  username: "username",
  currentHabits: [
    { name: "write", frequency: "1", type: "health", checked: [] }
  ],
  previousHabits: [],
  currentWeek: "y1w1",
  previousWeek: "y1w0",
  onAddNewHabit: () => {},
  onUpdateHabit: () => {},
  toggleDayHabit: () => {}
};

jest.mock("@rafael.audy/habit-geek-utils/utils/dateUtils.js", () => ({
  getTodayIndex: jest.fn().mockReturnValue(1),
  getCurrentWeek: jest.fn().mockReturnValue("y1w1"),
  getWeekIntervalText: jest.fn().mockReturnValue("30/Sep ... 6/Oct"),
  isToday: jest.fn().mockReturnValue(false)
}));

describe("DashboardView component", () => {
  it("Renders static elements", () => {
    const dashboardView = shallow(<DashboardView {...props} />);
    expect(dashboardView).toMatchSnapshot();
  });

  it("Doesn't show username is none is provided", () => {
    const dashboardView = shallow(<DashboardView {...props} username="" />);
    expect(dashboardView.find("h2")).toMatchSnapshot();
  });

  it("Doesn't show the add button on previous weeks", () => {
    const dashboardView = mount(<DashboardView {...props} username="" />);
    dashboardView
      .find("button")
      .at(0)
      .simulate("click");
    expect(dashboardView.find(".btn-primary").length).toBe(0);
  });

  it("starts to create a habit", () => {
    const onAddNewHabitMock = jest.fn();
    const dashboardView = shallow(
      <DashboardView {...props} onAddNewHabit={onAddNewHabitMock} />
    );
    dashboardView.find(".btn-primary").simulate("click");
    expect(onAddNewHabitMock).toHaveBeenCalledTimes(1);
  });
});
