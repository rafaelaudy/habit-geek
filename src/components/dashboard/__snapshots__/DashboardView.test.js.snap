// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardView component Doesn't show username is none is provided 1`] = `
<h2
  className="mb-3"
>
  my-habits-title
</h2>
`;

exports[`DashboardView component Renders static elements 1`] = `
<div
  className="large-size-container"
>
  <h2
    className="mb-3"
  >
    my-habits-title
  </h2>
  <DashboardWeekHeader
    hasPreviousWeek="y1w0"
    isCurrentWeek={true}
    toggleWeek={[Function]}
    week="y1w1"
  />
  <HabitTable
    habits={
      Array [
        Object {
          "checked": Array [],
          "frequency": "1",
          "name": "write",
          "type": "health",
        },
      ]
    }
    onUpdateHabit={[Function]}
    toggleDayHabit={[Function]}
    week="y1w1"
  />
  <hr />
  <div
    className="d-flex justify-content-center"
  >
    <button
      className="btn btn-primary btn-lg"
      onClick={[Function]}
    >
      my-habits-new-habit
    </button>
  </div>
</div>
`;
